src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Legacy/Runtime.cs

```diff
@@ -107,7 +107,7 @@ public static void AssertNotDisposed(this JSObject self)
 
         public static void AssertInFlight(this JSObject self, int expectedInFlightCount)
         {
-            if (self.InFlightCounter != expectedInFlightCount) throw new InvalidProgramException($"Invalid InFlightCounter for JSObject {self.JSHandle}, expected: {expectedInFlightCount}, actual: {self.InFlightCounter}");
+            if (self.InFlightCounter != expectedInFlightCount) throw new InvalidProgramException(SR.Format(SR.UnsupportedLegacyMarshlerType, self.JSHandle, expectedInFlightCount, self.InFlightCounter));
```

Comment:
This sounds like a debug message. Is this user facing?
If does need to be localized, then use `ArgumentException`